services/core-api/app/api/mines/reports/models/mine_report.py

```diff
@@ -1,43 +1,97 @@
 import uuid
-from flask import current_app
 
 from sqlalchemy.dialects.postgresql import UUID
 from sqlalchemy.schema import FetchedValue
 from sqlalchemy.ext.associationproxy import association_proxy
+from sqlalchemy.ext.hybrid import hybrid_property
+from sqlalchemy.orm import column_property
+from sqlalchemy import select, and_, desc, asc
+from app.api.mines.reports.models.mine_report_submission import MineReportSubmission
+from app.api.mines.reports.models.mine_report_submission_status_code import MineReportSubmissionStatusCode
 
-from app.api.utils.models_mixins import Base, AuditMixin
 from app.extensions import db
+from app.api.utils.models_mixins import Base, AuditMixin
 from app.api.utils.include.user_info import User
 
 
 class MineReport(Base, AuditMixin):
     __tablename__ = "mine_report"
     mine_report_id = db.Column(db.Integer, primary_key=True, server_default=FetchedValue())
-    mine_report_guid = db.Column(UUID(as_uuid=True), server_default=FetchedValue())
+    mine_report_guid = db.Column(UUID(as_uuid=True), server_default=FetchedValue(), nullable=False)
+
     mine_report_definition_id = db.Column(
-        db.Integer, db.ForeignKey('mine_report_definition.mine_report_definition_id'))
+        db.Integer,
+        db.ForeignKey('mine_report_definition.mine_report_definition_id'),
+        nullable=False)
     mine_report_definition = db.relationship('MineReportDefinition', lazy='joined')
-    mine_guid = db.Column(UUID(as_uuid=True), db.ForeignKey('mine.mine_guid'))
+    mine_report_definition_guid = association_proxy('mine_report_definition',
+                                                    'mine_report_definition_guid')
+    report_name = association_proxy('mine_report_definition', 'report_name')
+    mine_report_category_xref = db.relationship(
+        'MineReportCategoryXref',
+        lazy='joined',
+        primaryjoin=
+        'MineReport.mine_report_definition_id==MineReportCategoryXref.mine_report_definition_id',
+        foreign_keys=mine_report_definition_id,
+        uselist=True,
+        order_by='MineReportCategoryXref.mine_report_category')
+    mine_report_category = association_proxy('mine_report_category_xref', 'mine_report_category')
```

Comment:
mine_report_categories should be a relationship using `secondary='mine_report_category_xref', don't need to traverse the xref table.

Reply:
Will talk to you in person.

Reply:
I will talk to you in person.